packages/discovery-react-components/src/components/DocumentPreview/utils/__tests__/documentData.test.ts

```diff
@@ -0,0 +1,55 @@
+import { getTextMappings } from '../documentData';
+import jsonDoc from '../../__fixtures__/Art Effects Koya Creative Base TSA 2008.pdf.json';
+
+const noTextMapping = {
+  extracted_metadata: {}
+};
+
+const invalidStringTextMapping = {
+  extracted_metadata: {
+    text_mappings:
+      'I have replaced to usual text mapping string here for a test. This should give an error.'
+  }
+};
+
+const noStringTextMapping = {
+  extracted_metadata: {
+    text_mappings: 1
+  }
+};
+
+describe('documentData', () => {
+  let consoleError: jest.SpyInstance<any, any>;
+
+  beforeAll(() => {
+    consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  afterAll(() => {
+    consoleError.mockRestore();
+  });
+
+  it('succeeds in getting text mapping from json file', () => {
+    const mappings = getTextMappings(jsonDoc);
+    expect(mappings).not.toEqual(null);
+    expect(consoleError).not.toHaveBeenCalled();
+  });
```

Comment:
Good test here of using real data. But I think you should also add "good" dummy data (with `text_mappings` as a JSON string) and specifically test that the returned data is an object with the expected properties.

Reply:
Went ahead and made those changes!